### PR TITLE
checkin for v9.2.1 build.

### PR DESCRIPTION
**Pull request by billysuh7**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- remove ppc arch as it is no longer supported in 9.2.1
- hardcode zlib version 1.2 because global pinning for zlib has become 1 recently
- drop sysroot, migrate to {{ stdlib('c') }} per https://github.com/conda-forge/conda-forge.github.io/issues/2102
- fix up tests as a lot of filenames got changed since 8.9.7.29
- use glibc 2.28 for cuda12 for aarch64

### CODE REVIEW

**Comment by billysuh7** [comment]
@conda-forge-admin, please rerender

**Comment by billysuh7** [comment]
cc: @mnicely for vis

**Comment by jakirkham** [comment]
cc @adibbley (for review)

**Comment by kmittman** [comment]
There is a new header in cuDNN v9 `cudnn_graph.h` and new .so files
- `libcudnn_engines_precompiled.so`
- `libcudnn_engines_runtime_compiled.so`
- `libcudnn_graph.so`
- `libcudnn_heuristic.so`

Do these need to be explicitly listed?

**Comment by jakirkham** [comment]
Thanks Billy and Kevin! 🙏

Looking at the build script it appears we are copying over content with globs. Are these lines sufficient or do they need to be adapted?

https://github.com/conda-forge/cudnn-feedstock/blob/523ec11359426e85da1ab24f6cade197e04799ec/recipe/meta.yaml#L32-L43

The other thing we may want to do is update the tests to check for these files

https://github.com/conda-forge/cudnn-feedstock/blob/6031c2a02e638ceb4fda5b606fbfe16de2dc6ce0/recipe/meta.yaml#L73-L83

**Comment by billysuh7** [comment]
@jakirkham @kmittman presumably globbing was being done as cuDNN has a few dozen files - should we strive to handle all of them ? I had simply replaced some files in the test section to suite v9.2.1 after files have changed, so it is still a certain selection of files. Do we have a preference or policy ? Thanks!

**Comment by kmittman** [comment]
LGTM

**Review by adibbley — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

Billy assuming you are referring to globbing in install? Agree this is reasonable to do. We can continue to expand testing to make sure we have the right things in the right places

https://github.com/conda-forge/cudnn-feedstock/blob/3ba3bf612108ddce0bf05d2e3abdad6451795429/recipe/meta.yaml#L29-L40

**Comment by billysuh7** [comment]
> Thanks all! 🙏
> 
> Billy assuming you are referring to globbing in install? Agree this is reasonable to do. We can continue to expand testing to make sure we have the right things in the right places
> 
> https://github.com/conda-forge/cudnn-feedstock/blob/3ba3bf612108ddce0bf05d2e3abdad6451795429/recipe/meta.yaml#L29-L40

Yes globbing in install in the case of cudnn which has many files. Thanks!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -48,21 +45,18 @@ build:
...
-    # available on conda-forge
-    #- sysroot_{{ target_platform }} 2.28  # [aarch64]
   host:
     - cuda-version 11.0  # [(cuda_compiler_version or "").startswith("11")]
     - cuda-version 12.0  # [(cuda_compiler_version or "").startswith("12")]
     - patchelf >=0.12  # [linux]
     # to suppress ".so not found" errors
-    - libzlib {{ zlib }}       # [linux]
-    - libzlib-wapi {{ zlib }}  # [win64]
+    - libzlib 1.2       # [linux]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:58]: why were these zlib put in place. this kinda goes against conda-forge's best practices.

why not just have

```
- zlib
```

Comment (billysuh7) [reply recipe/meta.yaml:58]: Thanks @hmaarrfk for pointing this out. This was done when I didn't have greater understanding of conda-forge (and I am still learning!). I had noticed the global pinning had been changed from 1.2 to 1, so I wrongly thought it was a downgrade, and so I wanted to stick with 1.2. Now I understand that 1 means 1.* so it covers 1.3.1 which is the latest and great version as of now. Going forward, I will stick w/ the previous {{ zlib }} to depend on global pinning. If there is no strong objection I am inclined to leave it as-is for cudnn 9.2.1 ...

Comment (hmaarrfk) [reply recipe/meta.yaml:58]: ok my main concern was about the runtime requirement, but it seems to cover 

1.3.1
image

**Pull request merged**